src/wallet/index.ts

```diff
@@ -181,12 +224,100 @@ class Wallet {
   }
 
   /**
-   * Gets the classic address of the account this wallet represents.
+   * Gets the classic address of the account this wallet represents. This only is correct if this wallet corresponds
+   * to your [master keypair](https://xrpl.org/cryptographic-keys.html#master-key-pair). If this wallet represents a
+   * [regular keypair](https://xrpl.org/cryptographic-keys.html#regular-key-pair) this will provide an incorrect address.
    *
    * @returns A classic address.
    */
   public getClassicAddress(): string {
-    return deriveAddress(this.publicKey)
+    return this.classicAddress
+  }
+
+  /**
+   *  Decode a serialized transaction, remove the fields that are added during the signing process,
+   *  and verify that it matches the transaction prior to signing. This gives the user a sanity check
+   *  to ensure that what they try to encode matches the message that will be recieved by rippled.
+   *
+   * @param serialized - A signed and serialized transaction.
+   * @param tx - The transaction prior to signing.
+   * @throws A ValidationError if the transaction does not have a TxnSignature/Signers property, or if
+   * the serialized Transaction desn't match the original transaction.
+   */
+  // eslint-disable-next-line class-methods-use-this -- Helper for organization purposes
```

Comment:
If you don't need `this` in the method, the function should be a private method outside of the class